narwhal/network/src/connectivity.rs

```diff
@@ -49,35 +55,54 @@ impl ConnectionMonitor {
         // TODO(metrics): Set `network_peers` to `connected_peers.len() as i64`
 
         // now report the connected peers
+        let mut peer_count: usize = connected_peers.len();
+        #[cfg(feature = "metrics")]
+        gauge!(snarkos_metrics::network::NETWORK_PEERS, peer_count as f64);
+
+        #[cfg(feature = "metrics")]
         for peer_id in connected_peers {
             self.handle_peer_connect(peer_id);
         }
 
         while let Ok(event) = subscriber.recv().await {
             match event {
                 anemo::types::PeerEvent::NewPeer(peer_id) => {
-                    self.handle_peer_connect(peer_id);
+                    _ = peer_id;
```

Comment:
If this is needed to silence an unused error, you could also do `_peer_id` and use the underscored var in the conditionally compiled function.

Reply:
I'm a bit torn here on what the most rust-esque approach here would look like. I'm not a fan of declaring variables as unused through a `_` prefix and then using them conditionally, so I chose to do the reverse and just tell Rust that the variable is to be marked as used in any case. Neither one is perfect, I just personally don't like the implied semantics of the first approach. Is this a hard opinion or are you fine with leaving it this way?